### PR TITLE
Added support for static methods and java utility classes

### PR DESCRIPTION
**Pull request by sachi-sharma**
Added support for static methods and java utility classes (System)

### CODE REVIEW

File: src/main/scala/edu/cmu/cs/obsidian/Main.scala [hunk 1]
```diff
@@ -287,27 +287,28 @@ object Main {
             for (p <- protobufs) {
                 val protobuf = p._1
                 val filename = p._2
+                if(filename.length() > 0) {
```
Comment (mcoblenz) [inline src/main/scala/edu/cmu/cs/obsidian/Main.scala]: Shouldn't we report an error if this is not the case?

File: src/main/scala/edu/cmu/cs/obsidian/codegen/TranslationContext.scala [hunk 2]
```diff
@@ -84,13 +84,17 @@ case class TranslationContext(
...
+        if(fieldLookup.keySet.contains(f)) {
+            (currentStateName, fieldLookup(f)) match {
+                /* same note in [assignVariable] above applies here as well */
+                case (None, GlobalFieldInfo(_)) => JExpr._this().ref(f)
+                case (Some(_), GlobalFieldInfo(_)) => contractClass.staticRef("this").ref(f)
+                case (_, StateSpecificFieldInfo(_, getFunc, _)) => JExpr.invoke(getFunc)
+            }
         }
+        else
+            model.directClass(f).staticRef("out")
```
Comment (mcoblenz) [inline src/main/scala/edu/cmu/cs/obsidian/codegen/TranslationContext.scala]: Why is this hardcoded here?

File: src/main/scala/edu/cmu/cs/obsidian/protobuf/ProtobufGen.scala [hunk 3]
```diff
@@ -21,15 +21,19 @@ object ProtobufGen {
 
         val protobufs: Seq[(Protobuf, String)] = program.imports.map ((imp: Import) => {
             // Each import results in a .proto file, which needs to be compiled.
-            val protobufOuterClassName = Util.protobufOuterClassNameForFilename(imp.name)
-            val protobufFilename = protobufOuterClassName + ".proto"
+            if(!imp.name.contains("/java-utilities/")) {
```
Comment (mcoblenz) [inline src/main/scala/edu/cmu/cs/obsidian/protobuf/ProtobufGen.scala]: Refactor this to pull out the constant.

File: src/main/scala/edu/cmu/cs/obsidian/typecheck/Checker.scala [hunk 4]
```diff
@@ -466,8 +466,18 @@ class Checker(globalTable: SymbolTable, verbose: Boolean = false) {
...
                      case (None, None) =>
-                         logError(e, VariableUndefinedError(x, context.thisType.toString))
-                         (BottomType(), context)
+                         val tableLookup = context.contractTable.lookupContract(x)
+                         if (!tableLookup.isEmpty) {
+                           val ctTableOfConstructed = tableLookup.get
+                           if(ctTableOfConstructed.contract.isInterface)
+                                (InterfaceContractType(ctTableOfConstructed), context)
+                           else
+                               (BottomType(), context)
```
Comment (mcoblenz) [inline src/main/scala/edu/cmu/cs/obsidian/typecheck/Checker.scala]: What does this mean? Why bottom type here?

Comment (sachi-sharma) [reply src/main/scala/edu/cmu/cs/obsidian/typecheck/Checker.scala]: Removed interface validation, as it is not relevant here.
- Added static validation in Checker instead

File: src/main/scala/edu/cmu/cs/obsidian/protobuf/ProtobufGen.scala (hunk 3 above)
Comment (sachi-sharma) [reply src/main/scala/edu/cmu/cs/obsidian/protobuf/ProtobufGen.scala]: Added constant

File: src/main/scala/edu/cmu/cs/obsidian/codegen/TranslationContext.scala (hunk 2 above)
Comment (sachi-sharma) [reply src/main/scala/edu/cmu/cs/obsidian/codegen/TranslationContext.scala]: Added constant

File: src/main/scala/edu/cmu/cs/obsidian/Main.scala (hunk 1 above)
Comment (sachi-sharma) [reply src/main/scala/edu/cmu/cs/obsidian/Main.scala]: Removed code generation for utility classes

**Pull request merged**